modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3612,7 +3612,8 @@ void GDScriptAnalyzer::reduce_call(GDScriptParser::CallNode *p_call, bool p_is_a
 			}
 		} else if (!is_self && base_type.is_meta_type && !p_call->is_static) {
 			base_type.is_meta_type = false; // For `to_string()`.
-			push_error(vformat(R"*(Cannot call non-static function "%s()" on the class "%s" directly. Make an instance instead.)*", p_call->function_name, base_type.to_string()), p_call);
+			const String base_class_name = base_type.to_string();
+			push_error(vformat(R"*(Cannot call non-static function "%s()" on the class "%s" directly. Make an instance using "var %s = %s.new()" and call functions on this instance.)*", p_call->function_name, base_class_name, base_class_name.to_snake_case(), base_class_name), p_call);
```

Comment:
My concern with being this explicit is that some users will take it literally and call `new()` without arguments for constructors that do require arguments.

Reply:
I suppose we can add `...` between the parentheses to denote that arguments that may be required. However, none of the built-in Godot classes require a parameter for `new()`, so maybe we can detect this and append the `...` conditionally.

Reply:
Also, this warning doesn't work correctly for unnamed classes, suggesting invalid syntax (`Make an instance using "var res://node.gd = res://node.gd.new()"`). I think we should just add a static example like `SomeClass.new()` rather than trying to implement smart hints, as that requires more effort than it looks.